models/intel/noise-suppression-poconetlike-0001/README.md

```diff
@@ -0,0 +1,56 @@
+# noise-suppression-poconetlike-0001
+
+## Use Case and High-Level Description
+
+This is a PoCoNet like model for noise suppression to make speech cleaner.
+The model is based on [PoCoNet](https://arxiv.org/abs/2008.04470) architecure and trained on clean and noise samples from [DNS-Challenge dataset](https://github.com/microsoft/DNS-Challenge/blob/master/README.md#dataset-licenses) [paper](https://arxiv.org/abs/2101.01902).
+The network works with mono audio sampled on 16kHz.
+The audio processed iterative by patches with 2048 size.
+On each iteration it takes 2048 (128ms) samples as input and returns 2048 (128ms) samples as output with 640 (40ms) samples delay.
+In addition the network required 52 state tensors to make processing seamless.
+On the first iteration these state tensors have to be filled by 0.
+On the consequences iterations theses tensors have to be taken from corresponding outputs of previous iteration.
+You can try [Noise Suppression Python\* Demo](../../../demos/noise_suppression_demo/python/README.md) to see how it works.
+
+## Specification
+
+to process 2048 samples that is 128ms for 16kHz
+
+| Metric            | Value                 |
+|-------------------|-----------------------|
+| GOps              | 1.2                   |
+| MParams           | 7.22                  |
+| Source framework  | PyTorch\*             |
+## Accuracy
+
+The [SISDR](https://arxiv.org/abs/1811.02508) quality metric was calculated on the 100 [dev test synthetic speech clips from DNS-Challenge dataset](https://github.com/microsoft/DNS-Challenge/tree/icassp2021-final/datasets/ICASSP_dev_test_set/track_1/synthetic).
+
+
+| Metric                          | Value         |
+|---------------------------------|---------------|
+| SISDR for input noisy signal    |    11.73   dB |
+| SISDR for output cleaned signal |    20.23   dB |
+| SISDR increase                  |    +8.49   dB |
+
+
+## Input
+
+Sequence patch, name: `input`, shape: `1, 2048`, format: `B, T`, where:
+
+ - `B` - batch size
+ - `T` - number of samples in patch
+
+input states, names: `inp_state_*`, should be filled by corresponding `out_state_*` from previous step
```

Comment:
Could all the `inp_state_*` be concatenated into one and unpacked inside of onnx?

May be this will remove a need in a workaround for demo. Out of curiosity, what does speed up blobs repacking in demo?
https://github.com/openvinotoolkit/openvino/blob/ae9f3ebc5d0c64b07780f653d3bf60685a6db5de/inference-engine/ie_bridges/python/src/openvino/inference_engine/ie_api.pyx#L1273
`_fill_inputs()` does the same, but it reuses blobs.

Reply:
"Could all the inp_state_* be concatenated into one and unpacked inside of onnx?"
Theoretically yes. I have not done this because:
1.  I believed that keep it separated speedup process. No need to pack-unpack and I did not expected that real copy is required to pass outputs blobs to input. And I still don't understand why it is needed to copy data.
2. The shape of packed blob will be [1200000]. Such unusual big size could raise unexpected problems. I tried to avid this.
3. It is harder to implement training with pack-repack. Model constantly changed due research period and such pack-unpack could be additional source of mistakes


The workaround in the demo gives 10x speedup.

Reply:
... I believe that the right way to work with such state tensors described here https://docs.openvinotoolkit.org/latest/openvino_docs_IE_DG_network_state_intro.html
But I don't know how to declare such states in pytorch. Will investigate it.

Reply:
1. I don't know for sure about speed of repacking: sending a single object between devices can be faster than many small ones. Similar thing was done for humanpose by @druzhkov-paul, but he also shrunk the amount of actual data.
Copying is safer in general: it allows `ndarray`s to be destroyed during inference.
2. That would be useful to know for OpenVINO if there is such an issue
3. I was thinking about doing that as last finalizing step. But I can understand, that you never know when you are actually done :)

> The workaround in the demo gives 10x speedup.

Yes, I already tried it myself, but do you know what causes the difference? The phrase "I did not expected that real copy is required to pass outputs blobs to input." leaves a shade that current approach doesn't perform copy. In that case the comment `**Copy** inputs manually through InferRequest functionality to speedup` in the demo is missleading.